src/dmd/parse.d

```diff
@@ -2152,6 +2154,35 @@ final class Parser(AST) : Lexer
                             cppmangle = token.value == TOK.class_ ? CPPMANGLE.asClass : CPPMANGLE.asStruct;
                             nextToken();
                         }
+                        else if (token.value == TOK.string_) // extern(C++, "namespaces")
+                        {
+                            ignoreCppSymbols = true;
+                            idents = new AST.Identifiers();
+                            AST.StringExp stringExp = cast(AST.StringExp)parsePrimaryExp();
+                            char[] names = stringExp.toStringz();
+                            uint startIndex = 0;
+                            while(startIndex < names.length)
+                            {
+                                uint i = startIndex;
+                                foreach(ref char c ; names[startIndex .. $])
+                                {
+                                    i += 1;
+                                    if (c == '.')
+                                    {
+                                        if (i != names.length)
+                                        {
+                                            c = '\0';
```

Comment:
Don't do this.

Reply:
Hell even this explicitly uses `'\0'`.
https://github.com/dlang/dmd/blob/v2.082.0/src/dmd/identifier.d#L200